controllers/backup-service/reconciler.go

```diff
@@ -0,0 +1,686 @@
+				err,
+			)
+		}
+
+		r.Log.Info("Created new Backup Service ConfigMap",
+			"name", getBackupServiceName(r.aeroBackupService))
+
+		return nil
+	}
+
+	r.Log.Info(
+		"ConfigMap already exist. Updating existing ConfigMap if required",
+		"name", getBackupServiceName(r.aeroBackupService),
+	)
+
+	desiredDataMap := make(map[string]interface{})
+	currentDataMap := make(map[string]interface{})
+
+	if err := yaml.Unmarshal(r.aeroBackupService.Spec.Config.Raw, &desiredDataMap); err != nil {
+		return err
+	}
+
+	data := cm.Data[BackupConfigYAML]
+
+	if err := yaml.Unmarshal([]byte(data), &currentDataMap); err != nil {
+		return err
+	}
+
+	currentDataMap["service"] = desiredDataMap["service"]
+	currentDataMap[common.BackupPoliciesKey] = desiredDataMap[common.BackupPoliciesKey]
+	currentDataMap[common.StorageKey] = desiredDataMap[common.StorageKey]
+	currentDataMap["secret-agent"] = desiredDataMap["secret-agent"]
+
+	updatedConfig, err := yaml.Marshal(currentDataMap)
+	if err != nil {
+		return err
+	}
+
+	cm.Data[BackupConfigYAML] = string(updatedConfig)
+
+	if err = r.Client.Update(
+		context.TODO(), cm, common.UpdateOption,
+	); err != nil {
+		return fmt.Errorf(
+			"failed to update Backup Service ConfigMap: %v",
+			err,
+		)
+	}
+
+	r.Log.Info("Updated Backup Service ConfigMap",
+		"name", getBackupServiceName(r.aeroBackupService))
+
+	return nil
+}
+
+func (r *SingleBackupServiceReconciler) getConfigMapData() map[string]string {
+	data := make(map[string]string)
+	data[BackupConfigYAML] = string(r.aeroBackupService.Spec.Config.Raw)
+
+	return data
+}
+
+func (r *SingleBackupServiceReconciler) reconcileDeployment() error {
+	var deploy app.Deployment
+
+	if err := r.Client.Get(context.TODO(),
+		types.NamespacedName{
+			Namespace: r.aeroBackupService.Namespace,
+			Name:      r.aeroBackupService.Name,
+		}, &deploy,
+	); err != nil {
+		if !errors.IsNotFound(err) {
+			return err
+		}
+
+		r.Log.Info("Create Backup Service deployment",
+			"name", getBackupServiceName(r.aeroBackupService))
+
+		deployment, err := r.getDeploymentObject()
+		if err != nil {
+			return err
+		}
+
+		// Set AerospikeBackupService instance as the owner and controller
+		err = controllerutil.SetControllerReference(
+			r.aeroBackupService, deployment, r.Scheme,
+		)
+		if err != nil {
+			return err
+		}
+
+		err = r.Client.Create(context.TODO(), deployment, common.CreateOption)
+		if err != nil {
+			return fmt.Errorf("failed to deploy Backup service deployment: %v", err)
+		}
+
+		return r.waitForDeploymentToBeReady()
+	}
+
+	r.Log.Info(
+		"Backup Service deployment already exist. Updating existing deployment if required",
+		"name", getBackupServiceName(r.aeroBackupService),
+	)
+
+	oldResourceVersion := deploy.ResourceVersion
+
+	desiredDeployObj, err := r.getDeploymentObject()
+	if err != nil {
+		return err
+	}
+
+	deploy.Spec = desiredDeployObj.Spec
+
+	if err = r.Client.Update(context.TODO(), &deploy, common.UpdateOption); err != nil {
```

Comment:
There are too many k8s mutated fields which won't be there in desiredSpec, so direct comparison won't be straight forward. Added direct update call here, in case of no spec change it will be a no op